File: app/src/main/java/in/aerem/comconbeacons/UsersPositionsAdapter.java
package in.aerem.comconbeacons;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

public class UsersPositionsAdapter extends RecyclerView.Adapter<UsersPositionsAdapter.ViewHolder> {
    private List<UserListItem> mDataset = new ArrayList<>();

    public void setData(List<UserListItem> newData) {
        this.mDataset = newData;
        notifyDataSetChanged();
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.recycler_view_item, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        UserListItem u = mDataset.get(position);
        holder.mUsernameView.setText(u.username);
        holder.mLocationView.setText(u.location);
        holder.mTimeView.setText(u.time);
    }

    @Override
    public int getItemCount() {
        return mDataset.size();
    }

    // Provide a reference to the views for each data item
    // Complex data items may need more than one view per item, and
    // you provide access to all the views for a data item in a view holder
    public static class ViewHolder extends RecyclerView.ViewHolder {
        // each data item is just a string in this case
        public TextView mUsernameView;
        public TextView mLocationView;
        public TextView mTimeView;

        public ViewHolder(View v) {
            super(v);
            mUsernameView = itemView.findViewById(R.id.username);
            mLocationView = itemView.findViewById(R.id.location);
            mTimeView = itemView.findViewById(R.id.time);
        }
    }
}

File: app/src/main/java/in/aerem/comconbeacons/UsersPositionsAdapter.kt
package `in`.aerem.comconbeacons

import `in`.aerem.comconbeacons.models.UserListViewModel
import `in`.aerem.comconbeacons.models.statusToResourceId
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import org.ocpsoft.prettytime.PrettyTime
import java.util.*

class UsersPositionsAdapter(private val mViewModel: UserListViewModel) : RecyclerView.Adapter<UsersPositionsAdapter.ViewHolder>() {
    private var mDataset: List<UserListItem> = ArrayList()

    fun setData(newData: List<UserListItem>) {
        this.mDataset = newData
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context)
            .inflate(R.layout.recycler_view_item, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val u = mDataset[position]
        holder.mStatusIconView.setImageResource(statusToResourceId(u.status.toString().toLowerCase()))
        holder.mUsernameView.text = u.username
        holder.mLocationView.text = u.location
        holder.mTimeView.text = PrettyTime(Locale("ru")).format(u.date)
        holder.mFavoriteButtonView.setImageResource(if (u.favorite) R.drawable.ic_star_24dp else R.drawable.ic_star_border_24dp)
        holder.mFavoriteButtonView.setOnClickListener { mViewModel.setFavorite(u.id, !u.favorite) }
    }

    override fun getItemCount(): Int {
        return mDataset.size
    }

    class ViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        var mStatusIconView: ImageView = itemView.findViewById(R.id.statusIcon)
        var mUsernameView: TextView = itemView.findViewById(R.id.username)
        var mLocationView: TextView = itemView.findViewById(R.id.location)
        var mTimeView: TextView = itemView.findViewById(R.id.time)
        var mFavoriteButtonView: ImageButton = itemView.findViewById(R.id.favoriteButton)
    }
}
